test/programs/type-intersection-recursive/main.ts

```diff
@@ -2,5 +2,5 @@ interface ChildFoo {
 }
 
 interface Foo {
-    readonly childFoos: Foo & ChildFoo;
+    readonly childFoos: Foo | ChildFoo;
```

Comment:
this type with &, is endless type, and no data to be valid for this type.

![image](https://github.com/YousefED/typescript-json-schema/assets/109151575/743e4891-febe-410a-8e02-e404a78ce5bf)